smartctl.go

```diff
@@ -399,6 +401,50 @@ func (smart *SMARTctl) mineDeviceErrorLog() {
 	}
 }
 
+func (smart *SMARTctl) mineDeviceSelfTest() {
+	validTypes := map[int]string{
+		255: "vendor",
+		129: "short_captive",
+		2:   "long",
+		1:   "short",
+	}
+
+	// assume the table will always be in descending order
+	processedTypes := make(map[string]bool)
+
+	for _, logEntry := range smart.json.Get("ata_smart_self_test_log.standard.table").Array() {
+		testType := int(logEntry.Get("type.value").Int())
+		testTime := float64(logEntry.Get("lifetime_hours").Int())
+		testRunningIndicator := int(logEntry.Get("status.value").Int())
+		testStatus := strconv.FormatBool(logEntry.Get("status.passed").Bool())
+
+		// stick with seconds
+		testTime = testTime * 60 * 60
+
+		// skip running tests
+		if testRunningIndicator != 0 {
```

Comment:
this is not correct, from one of my systems:
```
"status": {
"value": 41,
"string": "Interrupted (host reset)",
"remaining_percent": 90
}
```
`status.passeed` is NOT present in this case.

I don't have any SATA drives w/ failing checks to compare presentlyy, but I worry they are also non-zero.